src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -90,55 +134,120 @@ export default class CoExpressionTab extends React.Component<ICoExpressionTabPro
         );
     }
 
+    @computed get xOptions() {
+        if (this.props.store.coexpressionTabMolecularProfiles.isComplete) {
+            let filteredProfiles: MolecularProfile[] = [];
+            if (this.isSelectedGeneticEntityAGeneSet) {
+                filteredProfiles = getGenesetProfiles(this.props.store.coexpressionTabMolecularProfiles.result);
+            } else {
+                filteredProfiles = filterAndSortProfiles(this.props.store.coexpressionTabMolecularProfiles.result);
+            }
+            return filteredProfiles.map(profile=>{
+                const profiledSampleCount = this.props.store.molecularProfileIdToProfiledSampleCount.result![profile.molecularProfileId];
+                return {
+                    label: `${profile.name} (${profiledSampleCount} sample${profiledSampleCount !== 1 ? "s" : ""})`,
+                    value: profile.molecularProfileId
+                };
+            });
+        }
+    }
+
+    @computed get yOptions() {
```

Comment:
please turn all @computeds that wait for promises to be complete into remoteDatas - @alisman and I have discussed this and decided its the best solution even though sometimes it adds a bit of boilerplate

Reply:
sorry I missed that until now